core/arch/arm/plat-imx/imx_caam.c

```diff
@@ -90,6 +90,67 @@ static struct imx_caam_jr *caam_init_jr(struct imx_caam_ctrl *caam)
 	io_write32((vaddr_t)&caam->jrcfg[0].orsr, jr->njobs);
 	return jr;
 }
+
+TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	struct imx_caam_ctrl *caam = (struct imx_caam_ctrl *)
+		core_mmu_get_va(CAAM_BASE, MEM_AREA_IO_SEC);
+	struct imx_caam_jr *jr;
+	int ret = TEE_ERROR_SECURITY;
+	char outbuf[MKVB_SIZE] = { 0 };
+	uint32_t descriptor[4] = {
+		MKVB_DESC_HEADER,
+		MKVB_DESC_SEQ_OUT,
+		0,
+		MKVB_DESC_BLOB
+	};
+	paddr_t desc_paddr = 0;
+	int counter = 0;
+
+	caam_enable_clocks(true);
+	jr = caam_init_jr(caam);
+
+	descriptor[2] = virt_to_phys(&outbuf);
+	cache_operation(TEE_CACHEFLUSH, &outbuf, MKVB_SIZE);
+
+	cache_operation(TEE_CACHEFLUSH, &descriptor, 16);
+	desc_paddr = virt_to_phys(&descriptor);
+	memcpy(jr->inring, &desc_paddr, 4);
+	cache_operation(TEE_CACHEFLUSH, jr->inring,
+			jr->njobs*sizeof(struct imx_caam_inring_entry) + 4);
+	cache_operation(TEE_CACHEFLUSH, jr->outring,
+			jr->njobs*sizeof(struct imx_caam_outring_entry) + 4);
+
+	// Tell CAAM that one job is available
+	io_write32((vaddr_t)&caam->jrcfg[0].irjar, 1);
+
+	while (io_read32((vaddr_t)&caam->jrcfg[0].orsfr) != 1) {
+		dsb();
+		isb();
+		counter++;
+		if (counter > 1000)
+			break;
+	}
+
+	cache_operation(TEE_CACHEINVALIDATE, jr->outring,
+			sizeof(struct imx_caam_outring_entry));
+	cache_operation(TEE_CACHEINVALIDATE, &outbuf, MKVB_SIZE);
```

Comment:
This sequence invalidates cache lines related to the stack and to the heap. Hazardous regarding buffer alignment.